scala/e2e-tests/src/test/scala/uk/gov/nationalarchives/e2etests/StepDefs.scala

```diff
@@ -0,0 +1,193 @@
+package uk.gov.nationalarchives.e2etests
+
+
+import cats.effect.{Async, Clock, Ref}
+import cats.effect.std.AtomicCell
+import cats.syntax.all.*
+import fs2.Stream
+import io.circe.Decoder
+import io.circe.generic.auto.*
+import io.circe.syntax.*
+import org.apache.commons.codec.digest.DigestUtils
+import org.reactivestreams.FlowAdapters
+import org.typelevel.log4cats.{Logger, SelfAwareStructuredLogger}
+import org.typelevel.log4cats.slf4j.Slf4jFactory
+import pureconfig.*
+import pureconfig.generic.derivation.default.*
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue
+import software.amazon.awssdk.transfer.s3.model.CompletedUpload
+import uk.gov.nationalarchives.DADynamoDBClient.DADynamoDbWriteItemRequest
+import uk.gov.nationalarchives.e2etests.StepDefs.{Config, *}
+import uk.gov.nationalarchives.{DADynamoDBClient, DAS3Client, DASFNClient, DASQSClient}
+
+import java.net.URI
+import scala.concurrent.duration.*
+import java.time.{Instant, LocalDate}
+import java.util.UUID
+import scala.reflect.ClassTag
+import scala.util.Random
+
+class StepDefs[F[_]: Async](messageUuidCell: AtomicCell[F, List[SqsMessage]]) {
+
+  val s3Client: DAS3Client[F] = DAS3Client[F]()
+  val sqsClient: DASQSClient[F] = DASQSClient[F]()
+  val dynamoClient: DADynamoDBClient[F] = DADynamoDBClient[F]()
+  val sfnClient: DASFNClient[F] = DASFNClient[F]()
+  val config: Config = ConfigSource.default.loadOrThrow[Config]
+  val ingestComplete = "preserve.digital.asset.ingest.complete"
+  val ingestUpdate = "preserve.digital.asset.ingest.update"
```

Comment:
Maybe `ingestUpdateMessage` or `ingestUpdateMsg`?

Reply:
See above

Reply:
I can't use ExternalUtils easily because it's in a seperate subproject with its own build sbt. I quite like that it's seperate though, if we change it the tests will fail which feels right.